### PR TITLE
Patatrack integration - common data formats (5/N)

### PR DESCRIPTION
**Pull request by fwyzard**
#### PR description:

Common data formats and dictionaries for CUDA modules.

#### PR validation:

Changes in use in the Patatrack releases.

#### if this PR is a backport please specify the original PR and why you need to backport that PR:

Includes changes from:
  - cms-patatrack#384: Port the whole pixel workflow to new heterogeneous framework
  - cms-patatrack#385: Implement full Pixel SoA workflow on CPU
  - cms-patatrack#364: Move event and stream caches, and caching allocators out from CUDAService
  - cms-patatrack#389: Replace use of API wrapper stream and event with plain CUDA, part 1
  - cms-patatrack#416: Drop obsolete heterogenous framework
  - cms-patatrack#429: Implement changes from the CUDA framework review
  - cms-patatrack#442: Integrate the comments from the upstream PRs
  - cms-patatrack#470: Update HCAL local reconstruction on GPUs
  - cms-patatrack#526: Apply code formatting

### CODE REVIEW

**Comment by fwyzard** [comment]
@cmsbuild, please test

**Comment by fwyzard** [comment]
Looks the **HeaderConsistency** check does not understand CUDA ?

**Comment by makortel** [comment]
> Looks the **HeaderConsistency** check does not understand CUDA ?

Indeed. On the other hand the problem has already slipped into `master`, so I would not hold this PR because of that. But we should figure out how deal with that (e.g. should all headers with CUDA code be protected with `#ifdef`s? if yes, how would that play together with `cudaCompat`?)

**Comment by makortel** [comment]
> The following packages do not have a category, yet:
> 
> CUDADataFormats/StdDictionaries
> Please create a PR for https://github.com/cms-sw/cms-bot/blob/master/categories_map.py to assign category

Done in https://github.com/cms-sw/cms-bot/pull/1393

File: CUDADataFormats/StdDictionaries/src/classes_def.xml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+<lcgdict>
+    <class name="std::vector<std::byte, cms::cuda::HostAllocator<std::byte, 0>>" />
```
Comment (makortel) [inline CUDADataFormats/StdDictionaries/src/classes_def.xml:2]: Could you remind me, the plan ẁas to eventually get rid of these, right?

(well, eventually likely `HostProduct` and `HeterogneeousSoA` will evolve)

**Comment by makortel** [comment]
+heterogeneous

**Comment by silviodonato** [comment]
+1

**Comment by silviodonato** [comment]
merge

**Pull request merged**